### ISSUE TITLE
KernelExplainer error message

### ISSUE THREAD

**Issue by ChrisKuoColumbiaU** [shap/shap#1043]
Here is my code:
import shap
shap.initjs()
rf_explainer = shap.KernelExplainer(rf.predict, X_test)
rf_shap_values = rf_explainer.shap_values(X_test)

They encountered the following new errors. X_test is a normal pandas data frame.

---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
~/opt/anaconda3/envs/AD/lib/python3.7/site-packages/tqdm/notebook.py in status_printer(_, total, desc, ncols)
     95             if total:
---> 96                 pbar = IProgress(min=0, max=total)
     97             else:  # No total? Show info style bar with no progress tqdm status

NameError: name 'IProgress' is not defined

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
<ipython-input-18-006e506f66f2> in <module>
      2 shap.initjs()
      3 rf_explainer = shap.KernelExplainer(rf.predict, X_test)
----> 4 rf_shap_values = rf_explainer.shap_values(X_test)

~/opt/anaconda3/envs/AD/lib/python3.7/site-packages/shap/explainers/kernel.py in shap_values(self, X, **kwargs)
    205         elif len(X.shape) == 2:
    206             explanations = []
--> 207             for i in tqdm(range(X.shape[0]), disable=kwargs.get("silent", False)):
    208                 data = X[i:i + 1, :]
    209                 if self.keep_index:

~/opt/anaconda3/envs/AD/lib/python3.7/site-packages/tqdm/notebook.py in __init__(self, *args, **kwargs)
    204         total = self.total * unit_scale if self.total else self.total
    205         self.container = self.status_printer(
--> 206             self.fp, total, self.desc, self.ncols)
    207         self.sp = self.display
    208 

~/opt/anaconda3/envs/AD/lib/python3.7/site-packages/tqdm/notebook.py in status_printer(_, total, desc, ncols)
    102             # #187 #451 #558
    103             raise ImportError(
--> 104                 "FloatProgress not found. Please update jupyter and ipywidgets."
    105                 " See https://ipywidgets.readthedocs.io/en/stable"
    106                 "/user_install.html")

ImportError: FloatProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html

**Comment by alexcoca** [comment]
This has nothing to do with the explainer, but it seems that the ipywidgets might have to be updated for the progress bar to work. Try `conda install -c conda-forge ipywidgets` in your `AD` environment, restart your notebook and try again.

**Comment by arturomoncadatorres** [comment]
Indeed, installing/updating `ipywidgets` solves this problem. I am wondering if then `shap` shouldn't make `ipywidgets` a required dependency

**Comment by Pythalex** [comment]
If you want to avoid the dependency and if you don't need the progress bar, use the argument silent=True in shap_values().
